src/hooks/useTxDetails.ts

```diff
@@ -0,0 +1,32 @@
+import { sameAddress } from '@/utils/addresses'
+import type { TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
+import { getTransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
+import useAsync from './useAsync'
+import useChainId from './useChainId'
+import useSafeInfo from './useSafeInfo'
+
+/**
+ * @returns TransactionDetails for transaction id in url
+ */
+export const useTxDetails = (txId?: string): [TransactionDetails | undefined, Error | undefined, boolean] => {
```

Comment:
Heads up: https://github.com/safe-global/web-core/pull/853 will require you tweak this hook a bit.